src/super_gradients/common/auto_logging/console_logging.py

```diff
@@ -0,0 +1,145 @@
+import os
+import sys
+from datetime import datetime
+from pathlib import Path
+from io import StringIO
+import atexit
+
+from super_gradients.common.environment.env_helpers import multi_process_safe, is_distributed
+
+
+FILE_BUFFER_SIZE = 10_000
+
+
+class BufferWriter:
+    """File writer buffer that opens a file only when flushing and under the condition that threshold buffersize was reached."""
+
+    def __init__(self, filename: str, buffer_size: int):
+        self.buffer = StringIO()
+        self.filename = filename
+        self.buffer_size = buffer_size
+        self.open = open
+
+    def write(self, data: str):
+        self.buffer.write(data)
+
+    def flush(self, force: bool = False):
```

Comment:
this implementation is not thread-safe.
you are replacing self.buffer without locking it first to writing.
so one thread might be writing to self.buffer while the other is dumping it.
you have to use 
```
# create a lock
lock = Lock()
# acquire the lock
lock.acquire()
# ...
# release the lock
lock.release()
```

Reply:
Mmhh I might be wrong on that point, but I think that self.buffer is not shared among the threads.

Also if I create a lock I need to pass it to my threads when creating the threads for it to be shared among the threads, no ?

Reply:
I pushed this c4569d78aff59d1acd0974e654d24019c3bd0e4a and it worked all fine on Coco dataset with DDP (x3).